Embedded objects fixes

LotteHofstede opened the pull request:
This PR closes #270 
I rewrote the `BaseResponseParsingService` so it's able to parse (paginated) embedded objects as expected. Also made sure the `RemoteDataBuildService` knows how to handle paginated embeds. Added model classes and serialization mapping for `BitstreamFormat` and `ResourcePolicy`, so they can already be parsed from the responses where they are embedded.

tdonohue commented on src/app/core/shared/resource-policy.model.ts:
```diff
@@ -0,0 +1,20 @@
+import { CacheableObject } from '../cache/object-cache.reducer';
+import { ResourceType } from './resource-type';
+
+export class ResourcePolicy implements CacheableObject {
```
This class should have comments to describe the various fields.  See, for example, `item.model.ts`

tdonohue commented on src/app/core/cache/models/normalized-resource-policy.model.ts:
```diff
@@ -0,0 +1,23 @@
+
+@mapsTo(ResourcePolicy)
+@inheritSerialization(NormalizedObject)
+export class NormalizedResourcePolicy extends NormalizedObject {
```
This class should have comments to describe it.  What is "normalized" about this ResourcePolicy object?

tdonohue commented on src/app/core/cache/models/normalized-bitstream-format.model.ts:
```diff
@@ -0,0 +1,35 @@
+
+@mapsTo(BitstreamFormat)
+@inheritSerialization(NormalizedObject)
+export class NormalizedBitstreamFormat extends NormalizedObject {
```
This Class should have comments to describe it & its properties. What is "normalized" about this BitstreamFormat object?

tdonohue commented on src/app/core/cache/it-to-uuid-serializer.ts:
```diff
@@ -0,0 +1,19 @@
+import { hasValue } from '../../shared/empty.util';
+
+export class IDToUUIDSerializer {
```
Again, comments needed to describe class.

tdonohue commented on src/app/+item-page/field-components/collections/collections.component.spec.ts:
```diff
@@ -0,0 +1,74 @@
+    });
+  });
+
+  describe('When the requested item request has succeeded', () => {
```
Should say "When the requested item request has failed".

LotteHofstede commented:
Thanks for you feedback, Tim. I somehow missed it before.

I added documentation to all requested files and also extended or corrected some of the model classes. The test description you mentioned was also fixed, thanks for noticing!

The normalized model classes represent the objects as they are returned by the REST endpoint. In a lot of cases, objects have references to one or more objects, which are flattened into a UUIDs for serialization. This is why they are called normalized objects. 
The 'normal' model classes represent the version of the object as it is used by the UI.
